File: 1.数组/3.移除元素/26.删除有序数组中的重复项.cpp
/*
 * @lc app=leetcode.cn id=26 lang=cpp
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
class Solution {
    public:
        int removeDuplicates(vector<int>& nums) {
            int l1 = 0, l2 = 1, f = 0;
            while (l2 < nums.size())
            {
                if (nums[l1] != nums[l2])
                {
                    nums[++ l1] = nums[l2];
                }
                l2 ++;
            }
            return l1 + 1;
        }
    };
// @lc code=end

class Solution {
    public:
        int removeDuplicates(vector<int>& nums) {
            int l1 = 0, l2 = 1, f = 0;
            nums[f ++] = nums[l1];
            while (l2 < nums.size())
            {
                while (l2 < nums.size() && nums[l1] == nums[l2])
                {
                    l1 ++, l2 ++;
                }
                if (l2 >= nums.size()) break;
                nums[f ++] = nums[l2];
                l1 ++, l2 ++;
            }
            return f;
        }
    };

File: 面经150/数组&字符串/26.删除有序数组中的重复项.py
#
# @lc app=leetcode.cn id=26 lang=python3
#
# [26] 删除有序数组中的重复项
#
from mytools import *
# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        idx = 0
        i = 0
        while i < n:
            nums[idx] = nums[i]
            idx += 1; i += 1
            while i < n and nums[i] == nums[i - 1]: i += 1
        return idx
# @lc code=end

